cpp/cmake/thirdparty/get_cutlass.cmake

```
@@ -42,6 +44,7 @@ function(find_and_configure_cutlass)
     GIT_TAG ${PKG_PINNED_TAG}
     GIT_SHALLOW TRUE
     OPTIONS "CUDAToolkit_ROOT ${CUDAToolkit_LIBRARY_DIR}"
+    PATCH_COMMAND ${CMAKE_COMMAND} -E env GIT_COMMITTER_NAME=rapids-cmake GIT_COMMITTER_EMAIL=rapids.cmake@rapids.ai ${GIT_EXECUTABLE} am -3 ${CMAKE_CURRENT_SOURCE_DIR}/patches/cutlass/build-export.patch
```

CMAKE_CURRENT_SOURCE_DIR is wrong here as it evaluates to the directory of the CMakeLists.txt and not this file. Need to use `CMAKE_CURRENT_FUNCTION_LIST_DIR` 

```suggestion
    PATCH_COMMAND ${CMAKE_COMMAND} -E env GIT_COMMITTER_NAME=rapids-cmake GIT_COMMITTER_EMAIL=rapids.cmake@rapids.ai ${GIT_EXECUTABLE} am -3 ${CMAKE_CURRENT_FUNCTION_LIST_DIR}/patches/cutlass/build-export.patch
```